pkg/scheduler/scheduler.go

```diff
@@ -160,39 +160,26 @@ type Config struct {
 
 // New create a new KubernetesScheduler
 func New(config Config) *KubernetesScheduler {
+	executorId := config.Executor.ExecutorId.GetValue()
+	executorUID, err := uid.Parse(executorId)
+	if err != nil {
```

Comment:
please return the error instead of fatally exiting here. i'm really trying to limit the places where fatally exiting the program is OK. i'm generally not convinced that doing so in a constructor is OK.